src/addon.js

```diff
@@ -46,13 +46,16 @@ class Addon extends EventEmitter {
     let self = this;
     if (!_.isPlainObject(params)) throw new Error('Params must be an object');
 
-    let method, endpoint, query;
-    ({ method, endpoint, query } = params);
+    let method, endpoint, query, data;
```

Comment:
Maybe I would have used `body` which sounds a bit more self-explanatory than `data` which sounds more generic to.